src/staticdata_utils.c

```diff
@@ -715,11 +715,13 @@ static int64_t write_dependency_list(ios_t *s, jl_array_t* worklist, jl_array_t
     size_t i, l = udeps ? jl_array_len(udeps) : 0;
     for (i = 0; i < l; i++) {
         jl_value_t *deptuple = jl_array_ptr_ref(udeps, i);
-        jl_value_t *dep = jl_fieldref(deptuple, 1);              // file abspath
-        size_t slen = jl_string_len(dep);
+        jl_value_t *depalias = jl_fieldref(deptuple, 5);        // file @depot alias
+        size_t slen = jl_string_len(depalias);
         write_int32(s, slen);
-        ios_write(s, jl_string_data(dep), slen);
+        ios_write(s, jl_string_data(depalias), slen);
         write_float64(s, jl_unbox_float64(jl_fieldref(deptuple, 2)));  // mtime
+        write_uint64(s, jl_unbox_uint64(jl_fieldref(deptuple, 3)));    // fsize
+        write_uint32(s, jl_unbox_uint32(jl_fieldref(deptuple, 4)));    // hash
         jl_module_t *depmod = (jl_module_t*)jl_fieldref(deptuple, 0);  // evaluating module
```

Comment:
I am not sure about the changes here. We could also have paths that are outside of the `@depot`.

Reply:
Sorry I don't understand.
File size and content hash are independent of the file's location, no?

Reply:
The changes above that is depalias? I would have expected us to store in the the abspath a `@depot/`

Reply:
> I would have expected us to store in the the abspath a @depot/

`depalias` will be a `@depot/...` path if the package is located in any of `DEPOT_PATH` when precompiling. If its not, then we just emit an abspath as before. So things outside the depot can't be relocated and caches will have to be recompiled, but otherwise it should work.

The string replacement itself is done in `base/loading.jl` when pushing to `_require_dependencies`.